scintillometry/io/psrfits/hdu.py

```diff
@@ -64,20 +90,44 @@ def location(self):
             return EarthLocation(self.header['ANT_X'],
                                  self.header['ANT_Y'],
                                  self.header['ANT_Z'], u.m)
-        except KeyError:
+        except (KeyError, TypeError):
             return None
 
+    @location.setter
+    def location(self, loc):
+        """ Location setter. input should be an Astropy EarthLocation"""
+        # TODO, for the space base observatory, Earth Location may not apply.
+        self.hdu.header['ANT_X'] = loc.x.to_value(u.m)
+        self.hdu.header['ANT_Y'] = loc.y.to_value(u.m)
+        self.hdu.header['ANT_Z'] = loc.z.to_value(u.m)
+
     @property
     def start_time(self):
         return (Time(self.header['STT_IMJD'], format='mjd', precision=9,
                      location=self.location) +
                 TimeDelta(self.header['STT_SMJD'], self.header['STT_OFFS'],
                           format='sec', scale='tai'))
 
+    @start_time.setter
+    def start_time(self, time):
+        """ Set the start_time, the input value should be an Time object"""
+        # Should we allow time set location
+        mjd_int = int(time.mjd)
+        mjd_frac = (time - Time(mjd_int, scale=time.scale, format=time.format))
+        frac_sec, int_sec = np.modf(mjd_frac.to(u.s).value)
+        self.hdu.header['STT_IMJD'] = '{0:05d}'.format(mjd_int)
+        self.hdu.header['STT_SMJD'] = '{}'.format(int(int_sec))
+        self.hdu.header['STT_OFFS'] = '{0:17.15f}'.format(frac_sec)
+        self.hdu.header['DATE-OBS'] = Time(time, precision=3).fits
```

Comment:
I'd just pass in a `precision=9` string - other codes should be able to handle it.

Reply:
precision=9 actually returns a longer digit of string which does not always follow the PSRFITS field. Plus this field is not meant to be very accurate.

Reply:
OK, I indeed see that the documentation doesn't even give any fractional seconds.